tests/vm/test_errors.ml

```diff
@@ -0,0 +1,128 @@
+open Lambda_vm
+
+module Testable = Vm_test.Testable
+
+module Compilation_and_execution_errors = struct
+  let check_execution_error result expected_error =
+    let open Vm_test in
+    match result with
+    | Error (Execution_error error) ->
+      Alcotest.(check Testable.execution_error)
+        "Same error" expected_error error
+    | Ok _ -> Alcotest.fail "Ast shouldn't execute"
+    | Error (Compilation_error error) ->
+      Alcotest.failf "%a" pp_compile_error error
+
+  let check_compilation_error result expected_error =
+    let open Vm_test in
+    match result with
+    | Error (Compilation_error error) ->
+      Alcotest.(check Testable.compilation_error)
+        "Same error" expected_error error
+    | Ok _ -> Alcotest.fail "Ast shouldn't execute"
+    | Error (Execution_error error) ->
+      Alcotest.failf "%a" pp_execution_error error
+
+  let test_compilation_undefined_variable () =
+    let script = [%lambda_vm.script fun _ -> x] in
+    check_compilation_error
+      (Vm_test.execute_ast 2000 (Int64 0L) script)
+      Undefined_variable
+
+  let test_fst_value_is_not_pair () =
+    let script = [%lambda_vm.script fun _ -> (fst 1L, (0L, 0L))] in
+    check_execution_error
+      (Vm_test.execute_ast 1201 (Int64 0L) script)
+      Value_is_not_pair
+
+  let test_snd_value_is_not_pair () =
+    let script = [%lambda_vm.script fun _ -> (snd 1L, (0L, 0L))] in
+    check_execution_error
+      (Vm_test.execute_ast 1201 (Int64 0L) script)
+      Value_is_not_pair
+
+  let test_op1_value_is_not_int64 () =
+    let script = [%lambda_vm.script fun _ -> (not (0L, 0L), (0L, 0L))] in
+    check_execution_error
+      (Vm_test.execute_ast 1601 (Int64 0L) script)
+      Value_is_not_int64
```

Comment:
Some op1 will work on pairs, like `fst` and `snd` so this test name is not very good